content/docs/user-guide/assets/runtime.md

```diff
@@ -0,0 +1,48 @@
+
+Note that the engine does not currently support load cancelling (cancelling a load request before completion), making it possible to call GetAsset and then discard the result while waiting for `OnAssetReady` to be called.  This is highly discouraged as it would stop functioning if/when load cancelling support is added.
+
+### Dependency Loading
+
+By default, the `AssetManager` will load all the dependencies of an asset before signalling an asset has completed loading.  Asset dependencies are recorded in the `AssetCatalog` and are determined by the Product Dependencies emitted by each builder.  There are 3 different load settings possible for dependencies (`AssetLoadBehavior`):
+1. Preload - The dependency will complete loading before the parent asset is signalled ready.
+2. QueueLoad - The dependency will start loading when the parent asset starts but will not be required to complete before the parent asset has been signalled ready.
+3. NoLoad - The dependency will not be requested to load.  If the dependency is already loaded anyway, any references will still point to the already-loaded asset.
+
+Keep in mind that regardless of the settings, AssetManager makes no guarantees on the order assets complete loading.
+
+### Hot-Reloading
+
+The Asset System provides support for hot-reloading, which is the updating of loaded, in-memory assets when the on-disk asset changes.  When the Asset Processor finishes processing an asset, a notification is sent out to the engine; if the asset in question is already loaded or loading, a reload request is queued and the Asset Manager starts loading the newly updated asset.  Each system utilizing the Asset System must implement support for this individually by handling the [OnAssetReloaded](https://github.com/o3de/o3de/blob/18205539abf1b1d2eb3959c0a1c42a3eea16a455/Code/Framework/AzCore/AzCore/Asset/AssetCommon.h#L603) event.
+
+## Asset Processing Prioritization
+
+The following are generally only relevant during project development, when assets aren't always processed ahead of time before running the engine.  In a release project where all assets have been compiled and bundled, these concepts do not apply.
+
+### Critical Assets
+
+Critical assets are assets which are required to start the engine and must be processed by the Asset Processor before start-up can complete.  Critical assets are prioritized by Asset Processor before any other type of asset.  They are declared by Asset Builders as part of the [Job Descriptor](https://github.com/o3de/o3de/blob/18205539abf1b1d2eb3959c0a1c42a3eea16a455/Code/Tools/AssetProcessor/AssetBuilderSDK/AssetBuilderSDK/AssetBuilderSDK.h#L446).  It is recommended to keep these type of assets to a minimum as they can considerably impact the time taken to start the engine.

```

Comment:
It might be worth adding a note that non-critical assets can process before or during critical asset processing if the non-critical asset is a job dependency of a critical asset.

We've had people bring that up before as a point of confusion.